packages/charting/src/components/LineChart/LineChart.base.tsx

```diff
@@ -235,9 +262,37 @@ export class LineChartBase extends React.Component<ILineChartProps, ILineChartSt
       };
       return legend;
     });
+
+    const colorFillBarsLegendDataItems = this.colorFillBars.map((colorFillBar: IColorFillBarsProps, index: number) => {
+      const title = colorFillBar.name;
+      const legend: ILegend = {
+        title,
+        color: colorFillBar.color,
+        action: () => {
+          if (this.state.selectedLegend === title) {
+            this.setState({ selectedLegend: '' });
+            this._handleColorFillBarLengendClick(colorFillBar, null);
+          } else {
+            this.setState({ selectedLegend: title });
+            this._handleColorFillBarLengendClick(colorFillBar, title);
+          }
+          this.setState({ activeLegend: title });
+        },
+        onMouseOutAction: () => {
+          this.setState({ activeLegend: '' });
+        },
+        hoverAction: () => {
+          this.setState({ activeLegend: title });
+        },
+        opacity: 0.4,
+        stripePattern: colorFillBar.applyPattern,
+      };
+      return legend;
+    });
+
     const legends = (
       <Legends
-        legends={legendDataItems}
+        legends={[...legendDataItems, ...colorFillBarsLegendDataItems]}
```

Comment:
i believe instead of appending here,  we can do this in the function `_createLegends`.. in this we we don't need to define
`colorFillBarsLegendDataItems ` function at all. because these are doing same things

Reply:
i am guessing , in this way we can also get the feature multi legend selection. with some minimal tweaks.

Reply:
This is in `_createLegends` and multi legend selection has been implemented